<launch>
    <!--@@HERE@@, change map name-->
    <arg name="map" default="1009_map"/>

    <env name="ROSCONSOLE_FORMAT" value="[${severity}][${thread}][${node}/${function}:${line}]: ${message}"/>
    <arg name="waypoints_topic" default="/waypoints"/>
    <arg name="addpose_topic" default="/add_pose"/>
    <arg name="addpose_rviz_topic" default="/initialpose"/>
    <arg name="operating_mode_topic" default="/operating_mode"/>
    <arg name="posearray_topic" default="/waypoint_poses"/>
    <arg name="distance_threshold" default="0.3"/>


    <node pkg="nav" type="main.py" name="follow_waypoints" output="screen" clear_params="true">
        <param name="goal_frame_id" value="map"/>
        <param name="waypoints_topic" value="$(arg waypoints_topic)"/>
        <param name="addpose_topic" value="$(arg addpose_topic)"/>
        <param name="addpose_rviz_topic" value="$(arg addpose_rviz_topic)"/>
        <param name="operating_mode_topic" value="$(arg operating_mode_topic)"/>
        <param name="posearray_topic" value="$(arg posearray_topic)"/>
        <param name="distance_tolerance" type="double" value="$(arg distance_threshold)"/>

    </node>

<node type="rviz" name="rviz" pkg="rviz" args="-d $(find nav)/rviz/nav.rviz" />

<!--ros navigation-->    
    <!-- Run the map server --> 
    <arg name="map_file" default="$(find amr_ui)/map_data/$(arg map).yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" respawn="true" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/> -->
        <remap from="cmd_vel" to="/ddmr_velocity_controller/cmd_vel"/>
        <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>        
        <rosparam file="$(find nav)/config/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
        <rosparam file="$(find nav)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find nav)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find nav)/config/global_costmap_params.yaml" command="load" /> 
        <rosparam file="$(find nav)/config/base_local_planner_params.yaml" command="load" />
    </node>
</launch>